Sources/Apollo/DataLoader.swift

```diff
@@ -1,52 +1,42 @@
 import Dispatch
 
 final class DataLoader<Key: Hashable, Value> {
-  public typealias BatchLoad = ([Key]) -> Promise<[Value]>
-  typealias Load = (key: Key, fulfill: (Value) -> Void, reject: (Error) -> Void)
-
+  public typealias BatchLoad = (Set<Key>) throws -> [Key: Value]
```

Comment:
Mostly because `DataLoader` uses a `Set` internally, and it seemed like a waste to force it to create a new array to pass it to the batch load function. Semantically it also makes sense I think, because keys should be unique. I looked into making the function generic or using a wrapper like `AnySequence` to hide the implementation, but that seemed like a lot of added complexity for no real gain.

Reply:
Oh, the other reason is we no longer rely on the values to be returned in order, because it now returns `[Key: Value]` instead. That also avoids the extra work #1519 made us aware of.

Reply:
Yeah agreed on not hiding the implementation, `Set` I think makes this way easier to understand. Thanks for the explanation!